app/celery/tasks.py

```diff
@@ -476,13 +480,40 @@ def update_letter_notifications_statuses(self, filename):
         for update in notification_updates:
             check_billable_units(update)
             update_letter_notification(filename, temporary_failures, update)
+            sorted_letter_counts[update.cost_threshold] += 1
 
         if temporary_failures:
             # This will alert Notify that DVLA was unable to deliver the letters, we need to investigate
             message = "DVLA response file: {filename} has failed letters with notification.reference {failures}".format(
                 filename=filename, failures=temporary_failures)
             raise DVLAException(message)
 
+        if sorted_letter_counts.keys() - {'Unsorted', 'Sorted'}:
+            unknown_status = sorted_letter_counts.keys() - {'Unsorted', 'Sorted'}
+
+            message = 'DVLA response file: {} contains unknown Sorted status {}'.format(
+                filename, unknown_status
+            )
+            raise DVLAException(message)
+
+        sent_at_date = get_sent_at_date_in_bst_from_filename(filename)
+        persist_daily_sorted_letter_counts(sent_at_date, sorted_letter_counts)
+
+
+def get_sent_at_date_in_bst_from_filename(filename):
```

Comment:
The date in the filename is the date that DVLA process the files - we've double checked and this is the date we want to store. But maybe this means that we need to change the names of some things instead, for example `notification_sent_day` since it isn't the day when the notification was sent.

Reply:
I agree, it would be confusing to leave the column name as `notification_sent_day`. I think `billing_day` will work.

Reply:
I've updated the column to be called `billing_day` now and changed the names of a couple of functions to make things clearer.